staking/availability/measure.go

```diff
@@ -186,14 +186,15 @@ type Reader interface {
 	ReadValidatorSnapshot(addr common.Address) (*staking.ValidatorWrapper, error)
 }
 
-// SetInactiveUnavailableValidators sets the validator to
+// ComputeAndRecord sets the validator to
 // inactive and thereby keeping it out of
 // consideration in the pool of validators for
 // whenever committee selection happens in future, the
 // signing threshold is 66%
-func SetInactiveUnavailableValidators(
-	bc Reader, state *state.DB, addrs []common.Address,
+func ComputeAndRecord(
```

Comment:
Why not move the logic here into the IncrementValidatorSigningCounts? Then you don't need to do Read and UpdateValidatorWrapper twice.

Reply:
Thought about it but it’s a separation of concerns. But sure can fold them both together